### PR TITLE
feat: return btc address during auth, closes #2909

### LINKED ISSUE #2909: [Bitcoin] return BTC address on auth
As requested by Friedger for catamaranswaps.

### LINKED ISSUE #3092: Return Bitcoin address to app during auth
This requires opening a SIP and issue in Connect for standardization

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/4252690108).

This PR adds returning of btc addresses during auth. 

I had to make [changes to `@stacks/wallet-sdk`](https://github.com/hirosystems/stacks.js/pull/1450) to support this. It follows a similar structure to `stxAddress` but for bitcoin, and nested by payment type.

```js
{
  btcAddress: {
    p2wpkh: {
      mainnet: "bc1qp2tmhmsgr4p9rtw3t2cm4za6xz5zkqukmf6a9c",
      testnet: "tb1qdlr74plsdz0ew5qdfqyr9da2nxqdqtsl4fmq5v"
    }
  }
}
```

I had to do a small refactor of the keychain code to be able to pull both networks, rather than just the current one.

https://user-images.githubusercontent.com/1618764/220606725-9d7935bf-9738-42b2-9ef6-231c42a7674f.mp4

 <br />

(Auth takes ages because of Gaia network requests ✨ )

cc/ @aulneau

### CODE REVIEW

**Review by edu-stx — APPROVED** [review]

**Comment by aulneau** [comment]
Rather than native segwit as the label, what do we think of p2wsh etc? This feels more accurate to me and flexible.

**Comment by kyranjamie** [comment]
Not against calling it `p2wpkh` either

**Comment by aulneau** [comment]
> Not against calling it `p2wpkh` either

Yeah I think I'd prefer this personally, feels more conventional and leaves the door open.

File: src/app/common/utils.ts [hunk 1]
```diff
@@ -312,6 +312,11 @@ export function whenStxChainId(chainId: ChainID) {
   return <T>(chainIdMap: WhenChainIdMap<T>): T => chainIdMap[chainId];
 }
 
+type NetworkMap<T> = Record<NetworkModes, T>;
+export function whenNetwork(mode: NetworkModes) {
```
Comment (fbwoolf) [inline src/app/common/utils.ts:321]: Maybe the `when...` helper functions should start being pulled out into their own file?

**Review by fbwoolf — APPROVED** [review]

File: src/app/common/utils.ts (hunk 1 above)
Comment (kyranjamie) [reply src/app/common/utils.ts:321]: I've been thinking about the common folder in general. Think we could organise better.

Perhaps at a higher level organising between:
- domain-specific logic (anything stx/btc related)
- language helpers (`isString` etc)
- ui helpers (`truncateMiddle`)

Comment (fbwoolf) [reply src/app/common/utils.ts:321]: Coming back here, I like this idea. I agree the `common` folder needs more structure. I'll make an issue for this if there isn't one already.

**Pull request merged**